services/src/main/java/org/keycloak/theme/ResourceLoader.java

```diff
@@ -0,0 +1,51 @@
+package org.keycloak.theme;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+
+public class ResourceLoader {
+
+    public static InputStream getResourceAsStream(String root, String resource) throws IOException {
+        if (root == null || resource == null) {
+            return null;
+        }
+        Path rootPath = Path.of(root);
```

Comment:
In the `getFile()` method this path is normalized first, should that be happening here as well?
```suggestion
        Path rootPath = Path.of(root).normalize();
```